docs/wiki/EnC-Supported-Edits.md

```diff
@@ -50,3 +50,24 @@ This document captures the current state. Potential future improvements in this
 | Edit a member referencing an embedded interop type | - |
 | Edit a member with On Error or Resume statements | Specific to Visual Basic |
 | Edit a member containing an Aggregate, Group By, Simple Join, or Group Join LINQ query clause | Specific to Visual Basic |
+| Edit in a solution containing projects that specify `*` in `AssemblyVersionAttribute`, e.g. `[assembly: AssemblyVersion("1.0.*")`]. | See [workaround](#projects-with-variable-assembly-versions) below. |
+
+### Projects with variable assembly versions
+
+Hot Reload and Edit & Continue are not compatible with using `*` in `AssemblyVersionAttribute` value. Presence of `*` in the version means that the compiler generates a new version
+every build based on the current time. The build then produces non-reproducible, non-deterministic outputs (see [Reproducible Builds](https://reproducible-builds.org)).
+
+It is thus highly recommended to use alternative versioning methods, such as [Nerdbank.GitVersioning](https://github.com/dotnet/Nerdbank.GitVersioning), that derive the assembly version
+from the HEAD commit SHA (for git repositories).
+
+> To enable generating commit SHA based assembly versions using `Nerdbank.GitVersioning` package, specify `{ "assemblyVersion" : {"precision": "revision"} }` setting in `version.json`.
```

Comment:
This is accurate, although I wonder if it's common for users to use wildcards with `AssemblyVersion` as opposed to `AssemblyFileVersion`. Having `AssemblyVersion` change with every single has at least historically been problematic (for strong named assemblies) so I would expect wildcards to be used for the file version instead. And in that case, this JSON isn't necessary, because `AssemblyFileVersion` by default includes a commit SHA component.

Reply:
We have seen customers using wildcard in assembly version, which is causing Hot Reload issues.